packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -766,14 +765,15 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
     ) => {
       switch (keyboardKey.getCode(e)) {
         case keyboardKey.Tab:
-          if (!_.isNil(highlightedIndex)) {
-            selectItemAtIndex(highlightedIndex)
-            // Keep focus here if condition applies.
-            if (this.props.multiple && !this.props.moveFocusOnTab) {
-              e.preventDefault()
+          if (this.state.open) {
```

Comment:
lets encapsulate this logic as it is identical to the one that is provided here: https://github.com/stardust-ui/react/pull/1150/files#diff-27df3be097c43ac2046615309f24d815R912. Also, it would be great to simplify this logic, as there is already lots of branching that becomes very hard to track

Reply:
you're right, will do that. thanks!